### PR TITLE
Refs #34233 -- Used aiter() and anext().

### PR DESCRIPTION
**Pull request by ngnpope**
Available since Python 3.10.

### CODE REVIEW

File: django/core/handlers/asgi.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ async def send_response(self, response, send):
             #   allow mapping of a sync iterator.
             # - Use aclosing() when consuming aiter.
             #   See https://github.com/python/cpython/commit/6e8dcda
-            async with aclosing(response.__aiter__()) as content:
+            async with aclosing(aiter(response)) as content:
```
Comment (felixxm) [inline django/core/handlers/asgi.py:271]: Should we update the comment above? :thinking: 
```
Consume via aiter() ...
```

Comment (ngnpope) [reply django/core/handlers/asgi.py:271]: Perhaps. I wasn't sure whether it was explicitly referring to the `.__aiter__()` of the response object when compared to `.streaming_content` on the same object. Feel free to choose and edit - I honestly don't know which to go for.

Comment (carltongibson) [reply django/core/handlers/asgi.py:271]: I think the comment should be left (it's as @ngnpope says)

**Review by carltongibson — APPROVED** [review]
Nice yes. 🏅

**Comment by felixxm** [comment]
@ngnpope Thanks :+1:

**Pull request merged**